packages/next/src/build/flying-shuttle/store-shuttle.ts

```diff
@@ -8,20 +12,57 @@ import {
   PAGES_MANIFEST,
   ROUTES_MANIFEST,
 } from '../../shared/lib/constants'
-import { recursiveCopy } from '../../lib/recursive-copy'
+
+export function getShuttleManifest(config: NextConfigComplete) {
+  const globalHash = crypto.createHash('sha256')
+
+  for (const key of Object.keys(process.env || {})) {
+    if (key.startsWith('NEXT_PUBLIC_')) {
+      globalHash.update(`${key}=${process.env[key]}`)
```

Comment:
do we need to check the value before we stringify it? ie to prevent stringifying undefined or something (not sure if that could happen here but noticed we do it in `getNextPublicEnvironmentVariables` in define-env-plugin)

Reply:
Ah missed that util, will update to use it